src/PluginClient.php

```diff
@@ -57,11 +57,43 @@ public function __construct($client, array $plugins = [], array $options = [])
             throw new \RuntimeException('Client must be an instance of Http\\Client\\HttpClient or Http\\Client\\HttpAsyncClient');
         }
 
-        $this->plugins = $plugins;
+        $this->plugins = $this->setPlugins($plugins);
         $this->options = $this->configure($options);
     }
 
     /**
+     * Append a plugin to the end of the queue.
+     *
+     * @param Plugin $plugin
+     */
+    public function addPlugin(Plugin $plugin)
```

Comment:
I would be ok with addPlugin in normal.

However, like described in our docs, order of plugins matters, so big :+1: for the appendPlugin / preprendPlugin method name.